impeller/renderer/backend/metal/surface_mtl.mm

```diff
@@ -127,7 +130,11 @@
     return false;
   }
 
-  [drawable_ present];
+  id<MTLCommandBuffer> command_buffer =
```

Comment:
The lock can fail in case of context loss. We need to check and return false from this method.

Reply:
Oh, derp. You got rid of the weak ptr. I think the weak ptr does make more sense though. We just need to be able to survive context loss.

Reply:
I just switched to hanging onto a `shared_ptr` instead.